marktplaats/query.py

```diff
@@ -133,11 +133,17 @@ def __init__(
             headers=REQUEST_HEADERS,
         )
 
-        # every request exception should raise here
+        # This catches HTTP 4xx and 5xx errors
         self.response.raise_for_status()
 
-        self.body = self.response.text
-        self.body_json = json.loads(self.body)
+        # But if it's something else non-200, still fail fast.
+        if self.response.status_code != 200:
+            raise Exception(f"Received non-200 status code", self.response)
```

Comment:
Could we make our own exception type for this? Someone may want to catch this error specifically.